src/pycram/ontology.py

```diff
@@ -0,0 +1,507 @@
+        else:
+            assert False, f'Ontology [{onto_.base_iri}]\'s name: {onto_.name} failed being loaded'
+
+        with onto_:
+            class OntologyConcept(Thing):
+                """
+                A default ontology concept class that inherits from owlready2.Thing with a list of designators as its attribute
+                """
+                namespace = onto_
+
+                def __init__(self, name: str):
+                    """
+                    Create a new ontology concept
+
+                    :param name: concept name
+                    """
+                    super().__init__(name)
+                    self.designators = []
+                    self.resolve = None
+
+                def get_default_designator(self) -> DesignatorDescription:
+                    """
+                    Return the first element of designators if there is, else None
+                    """
+                    return self.designators[0] if len(self.designators) > 0 else None
+
+    @classmethod
+    def save(cls):
+        """
+        Save the current ontology to disk
+        """
+        cls.onto.save(file=cls.onto_filename)
+        cls.onto_world.save()
+
+    @classmethod
+    def create_ontology_concept_class(cls, class_name: str,
+                                      onto_parent_concept_class: Optional[owlready2.Thing] = None)\
+            -> Type[owlready2.Thing]:
+        """
+        Create a new concept class in ontology
+
+        :param class_name: A given name to the new class
+        :param onto_parent_concept_class: An optional parent ontology class of the new class
+        :return: The created ontology class
+        """
+        return types.new_class(class_name, (cls.onto.OntologyConcept, onto_parent_concept_class,)
+                               if inspect.isclass(onto_parent_concept_class) else (cls.onto.OntologyConcept,))
+
+    @classmethod
+    def create_ontology_property_class(cls, class_name: str,
+                                       onto_parent_property_class: Optional[owlready2.Property] = None)\
+            -> Type[owlready2.Property]:
+        """
+        Create a new property class in ontology
+
+        :param class_name: A given name to the new class
+        :param onto_parent_property_class: An optional parent ontology property class of the new class
+        :return: The created ontology class
+        """
+        if onto_parent_property_class:
+            assert issubclass(onto_parent_property_class, owlready2.Property), \
+                f"{onto_parent_property_class} must be a subclass of one implementing owlready2.Property"
+        return types.new_class(class_name, (onto_parent_property_class,)
+                               if inspect.isclass(onto_parent_property_class) else (owlready2.ObjectProperty,))
+
+    @classmethod
+    def get_ontology_classes_by_condition(cls, condition: Callable, print_info=False, first_match_only=False, **kwargs)\
```

Comment:
printing or not should not be an argument of a function. This should be printed with the logger and the print statement should have the correct verbosity level, like info